railties/lib/rails/application.rb

```diff
@@ -257,13 +257,15 @@ def env_config
           "action_dispatch.backtrace_cleaner" => Rails.backtrace_cleaner,
           "action_dispatch.key_generator" => key_generator,
           "action_dispatch.http_auth_salt" => config.action_dispatch.http_auth_salt,
-          "action_dispatch.signed_cookie_salt" => config.action_dispatch.signed_cookie_salt,
-          "action_dispatch.encrypted_cookie_salt" => config.action_dispatch.encrypted_cookie_salt,
-          "action_dispatch.encrypted_signed_cookie_salt" => config.action_dispatch.encrypted_signed_cookie_salt,
-          "action_dispatch.authenticated_encrypted_cookie_salt" => config.action_dispatch.authenticated_encrypted_cookie_salt,
+          "action_dispatch.use_authenticated_cookie_encryption" => config.action_dispatch.use_authenticated_cookie_encryption,
+          "action_dispatch.primary_encryptor_scheme" => config.action_dispatch.primary_encryptor_scheme,
+          "action_dispatch.legacy_encryptor_schemes" => config.action_dispatch.legacy_encryptor_schemes,
+          "action_dispatch.primary_verifier_scheme" => config.action_dispatch.primary_verifier_scheme,
+          "action_dispatch.legacy_verifier_schemes" => config.action_dispatch.legacy_verifier_schemes,
           "action_dispatch.cookies_serializer" => config.action_dispatch.cookies_serializer,
-          "action_dispatch.cookies_digest" => config.action_dispatch.cookies_digest
-        )
+        ).tap do |config|
+          ActionDispatch::CookieSecurity.configure! config
```

Comment:
I had a feeling this was the case. I'm going to rework how the various configuration defaults are set, fix the dependency issues, and remove the top-level `CookieSecurity` constant.